direction/waiting/WaitingMiddle.cpp

```diff
@@ -0,0 +1,44 @@
+/**
+ * \file        WaitingMiddle.cpp
+ * \date        May 13, 2019
+ * \version     v0.8.1
+ * \copyright   <2009-2025> Forschungszentrum Jülich GmbH. All rights reserved.
+ *
+ * \section License
+ * This file is part of JuPedSim.
+ *
+ * JuPedSim is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * JuPedSim is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with JuPedSim. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * \section Description
+ * Implementation of a waiting strategy:
+ * Goal of this strategy is always the middle of the subroom or waiting area the pedestrian is in.
+ **/
+
+#include "WaitingMiddle.h"
+
+#include "geometry/Point.h"
+#include "geometry/Room.h"
+#include "geometry/SubRoom.h"
+#include "pedestrian/Pedestrian.h"
+
+Point WaitingMiddle::GetWaitingPosition(Room * room, Pedestrian * ped)
+{
+    SubRoom * subRoom = room->GetSubRoom(ped->GetSubRoomID());
```

Comment:
For the first if-else branch this `subRoom` is unused. Could you put this into the else branch?